modules/fts/pages/fts-troubleshooting.adoc

```diff
@@ -0,0 +1,244 @@
+* Experiment with boost to adjust this.
+For example, a query like "name:margaret^2 hobby:singing" would give twice as much weight to a match in the name field as the hobby field.
+
+== *Q: Why is indexing so slow?*
+
+By default, FTS tries to index all the fields in a document yet you typically only need to search on a few key fields.
+Indexing all fields requires more computation and results in larger indexes than you get if you limit the index to just the fields you need.
+
+Here are some tips to speed up indexing:
+
+* Build a custom mapping, set "dynamic": false to disable automatic indexing, and explicitly index only the named fields that you need to search.
+* Avoid numeric/date fields if you do not plan to query on these fields.
+* Avoid storing fields unless you absolutely need snippets or highlighting.
+* Consider disabling term vectors for a particular field if you know that you do not need snippets, highlights, or phrase search.
+
+== *Q: Why is querying so slow?*
+
+Slow queries can be caused by searches that require a lot of processing to satisfy.
+This is commonly due to a search includes high frequency terms, that is, terms that appear in many or even all documents.
+Searches on high frequency terms require more work than searches for low frequency terms and can easily be 100x slower due to the large number of matches.
+
+* If possible, consider adding high frequency terms to a stop word list and reindex so they are not excluded from the index altogether.
+High frequency words often contribute little to the search effectiveness and they can have a seriously detrimental impact on search performance, index size and index build times.
+* Check your indexes to rule out unexpected high frequency terms.
+Words like "the" and "a" in English are relatively obvious candidates to add to a stopword list, but you may also have unexpected high frequency terms in your index, perhaps because there is a common token in your data you didn’t expect or a tokenizer is breaking words in an unexpected way, leading to other common terms.
+* Try rewriting queries to be more efficient if you must search on high frequency terms.
+Using conjunction (AND) queries can help make the search faster.
+In a query string, you can do this by using the + operator (MUST).
+For If searching for the following is slow: lowFreqTerm highFreqTerm, try +lowFreqTerm highFreqTerm.
++
+Although it is a different search, it should perform better and may be acceptable.
+
+== *Q: Why are the queries getting rejected with HTTP status code 429?*
+
+At a high level, the HTTP status code 429 indicates that ftsMemoryQuota for the cluster is under-provisioned. Whenever Search service receives a query, it tries to forecast the memory requirements for performing that search request. And the query is rejected when fts service realises that the cumulative memory usage of the service goes beyond the set memory quota limit by executing the incoming query.
```

Comment:
And the query is rejected when fts service realises that the cumulative memory usage of the service goes beyond the set memory quota limit by executing the incoming query.

When processing an incoming query, the query gets rejected by the FTS service when the cumulative memory usage has exceeded the set memory quota limit.

Reply:
Change "fts service" to "FTS"

Reply:
Updated